### PR TITLE
Don't emit a bunch of warning logs during normal operation

### LINKED ISSUE #266: Prometheus should emit debug rather than warning when alert rules directory is missing
It should also only happen in install and upgrade events.

### PR DESCRIPTION
**Pull request by rbarry82**
## Issue
Various alert rules checks currently emit terrifying-looking warning messages on every invocation of charms, even when "must provide alert rules" is not a requirement.

Closes #266 

## Solution
Use logger.debug instead

## Testing Instructions
Deploy. Make sure there aren't scary warning logs if alert rules aren't present.

## Release Notes
Don't emit  warning logs during normal operation

### CODE REVIEW

**Review by sed-i — DISMISSED** [review]
Looks like we got a few fragile unit tests.

**Review by Abuelodelanada — APPROVED** [review]

File: tests/unit/test_endpoint_consumer.py [hunk 1]
```diff
@@ -436,9 +436,9 @@ def test_consumer_logs_an_error_on_missing_alerting_data(self):
         with self.assertLogs(level="WARNING") as logger:
             _ = self.harness.charm.prometheus_consumer.alerts()
             messages = logger.output
-            self.assertEqual(len(messages), 2)
+            self.assertEqual(len(messages), 1)
             self.assertIn(
-                "Alert rules were found but no usable group or identifier was present", messages[1]
+                "Alert rules were found but no usable group or identifier was present", messages[0]
```
Comment (sed-i) [inline tests/unit/test_endpoint_consumer.py:441]: Are those assertions needed at all?

**Review by sed-i — APPROVED** [review]

File: tests/unit/test_endpoint_consumer.py (hunk 1 above)
Comment (rbarry82) [reply tests/unit/test_endpoint_consumer.py:441]: I mean, it's `test_consumer_logs_an_error_on_missing_alerting_data`, so checking the logs is kinda crucial.

**Review by dstathis — APPROVED** [review]
:+1: Very much needed!

**Pull request merged**